sdk/dotnet/ApiManagement/ApiDiagnostic.cs

```diff
@@ -22,30 +22,34 @@ namespace Pulumi.Azure.ApiManagement
     /// 
     /// return await Deployment.RunAsync(() =&gt; 
     /// {
-    ///     var exampleResourceGroup = new Azure.Core.ResourceGroup("exampleResourceGroup", new()
+    ///     var example = new Azure.Core.ResourceGroup("example", new()
     ///     {
+    ///         Name = "example-resources",
```

Comment:
I believe that we intentionally omit `Name` because the resource is auto-named. My preference would be that we continue to drop auto-named fields (especially `Name`, which is likely to be structural), but we shouldn't block on it.

Reply:
It's not clear where such a pass would belong. PCL->Go should not care. HCL->PCL in the converter does not have enough metadata about whether a field is autonamed or not, or does it? Perhaps MarshallableDefaultInfo.AutoNamed?

Reply:
https://github.com/pulumi/pulumi-converter-terraform/issues/122 OK presumably this could be put in the converter.